BZ2070625: Encrypting EBS instance volumes with a KMS key (#48347)
openshift/openshift-docs: modules/storage-persistent-storage-volume-encrypt-with-kms-key.adoc

// Module included in the following assemblies:
//
// * storage/persistent_storage-aws.adoc

> [id="EBS-volume-encrypt_{context}"]

Reviewer: ```suggestion
[id="AWS-container-persistent-volumes-encrypt_{context}"]
```

Author: Addressed

> = Encrypting EBS instance volumes with a KMS key

Reviewer: ```suggestion
= Encrypting container persistent volumes on aws with a KMS key
```

Author: Addressed

.Prerequisites

* Storage must exist in the underlying infrastructure before you encrypt the EBS instance volumes in {product-title}.

.Procedure

> Defining a KMS key to encrypt EBS instance volumes is useful when you have explicit compliance and security guidelines when deploying to AWS.

Reviewer: ```suggestion
Defining a KMS key to encrypt container persistent volumes is useful when you have explicit compliance and security guidelines when deploying to AWS.
```

Author: Addressed

. Create a storage class:
+
[source,yaml]
----
$ cat << EOF | oc create -f -
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
name: <storage-class-name> <1>
parameters:

> fsType: ext4 <2>

Reviewer: add 1 parameter
```
encrypted: "true"
```

Author: Addressed

kmsKeyId: keyvalue <3>
provisioner: ebs.csi.aws.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
EOF
----
<1> Specifies the name of the storage class.
<2> File system that is created on provisioned volumes.

> <3> The full Amazon Resource Name (ARN) of the key to use when encrypting the volume. If none is supplied, but encypted is set to true, then AWS generates a key. See the link:https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html[AWS documentation] for a valid ARN value.

Reviewer: The parameter should be kms key ID and it could get from the aws offical doc: https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html

Author: Addressed

> . Create a persistent volume claim (PVC):

Reviewer: ```suggestion
. Create a persistent volume claim (PVC) with the storageClass which specify the kms key :
```

Author: Addressed

+
[source,yaml]
----
$ cat << EOF | oc create -f -
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: mypvc
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Filesystem

>   storageClassName: gp2-csi-kms

Reviewer: ```suggestion
  storageClassName: <storage-class-name>
```

Author: Addressed

// Module included in the following assemblies:
//
// * storage/persistent_storage-aws.adoc

> 

Reviewer 2: Can you set the content type to procedure here please?

Author: Addressed

[id="aws-container-persistent-volumes-encrypt_{context}"]

> = Encrypting container persistent volumes on AWS with a specific KMS key

Author: Addressed

.Prerequisites

> * Storage must exist in the underlying infrastructure before you encrypt the EBS instance volumes in {product-title}.

Reviewer 3: `before you encrypt the EBS instance volumes in {product-title}.` 
Since this is already the prerequisite section, is there something more implied here?

Author: Yes. Unless there is an underlying storage present, the customer wouldn't be able to encrypt the EBS volumes.

Author: Addressed

> * You must create a customer KMS key on AWS before you encrypt the container-persistent volumes in {product-title}.

Reviewer 3: `before you encrypt the container-persistent volumes in {product-title}.`
Redundant sentence like above

Author: Addressed

> * You must create a customer kms key on AWS before you encrypt the containers persistent volumes in {product-title}.

Reviewer 4: ```suggestion
* You must create a customer KMS key on AWS before you encrypt the containers persistent volumes in {product-title}.
```

> * You must create a customer kms key on AWS before you encrypt the containers persistent volumes in {product-title}.

Reviewer 4: ```suggestion
* You must create a customer KMS key on AWS before you encrypt the container's persistent volumes in {product-title}.
```

Author: Addressed

.Procedure

> Defining a KMS key to encrypt container-persistent volumes on AWS is useful when you have explicit compliance and security guidelines when deploying to AWS.

Reviewer 2: There shouldn't be an intro sentences to procedure steps (per the SSG), and this module is missing the intro paragraph. So the perfect solution would be to move this sentence up above the prerequisites to use as the intro paragraph. Can you make that move?

Author: Addressed

> Defining a KMS key to encrypt container persistent volumes on AWS is useful when you have explicit compliance and security guidelines when deploying to AWS.

Reviewer 4: See ISG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-hyphens#compound-words

> Defining a KMS key to encrypt container persistent volumes on AWS is useful when you have explicit compliance and security guidelines when deploying to AWS.

Author: Addressed

. Create a storage class:
+
[source,yaml]
----
$ cat << EOF | oc create -f -
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: <storage-class-name> <1>
parameters:
  fsType: ext4 <2>
  encrypted: "true"
  kmsKeyId: keyvalue <3>
provisioner: ebs.csi.aws.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
EOF
----
<1> Specifies the name of the storage class.
<2> File system that is created on provisioned volumes.

> <3> Specifies the full Amazon Resource Name (ARN) of the key to use when encrypting the container-persistent volume. If you do not provide any key, but the `encrypted` field is set to `true`, then the default KMS key is used. See the link:https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html[Finding the key ID and key ARN on AWS].

Reviewer 2: Can you remove "the" in the last sentence? You could also add "in the AWS documentation." if you cared to:

```suggestion
<3> Specifies the full Amazon Resource Name (ARN) of the key to use when encrypting the container-persistent volume. If you do not provide any key, but the `encrypted` field is set to `true`, then the default KMS key is used. See link:https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html[Finding the key ID and key ARN on AWS] in the AWS documentation.
```

Author: Addressed

> . Create a persistent volume claim (PVC) with the storage class specifying the KMS key :

Reviewer 2: Can you remove the extra space before the colon?

Author: Addressed

> . Create a persistent volume claim (PVC) with the storageClass specifying the KMS key :

Author: Addressed. 
storage class would be more accurate here

> <3> Specifies the full Amazon Resource Name (ARN) of the key to use when encrypting the container persistent volume. If you do not provide any key, but `encypted` field is set to `true`, then the default KMS key is used. See the link:https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html[Finding the key ID and key ARN on AWS].

Author: Addressed

> . Create a persistent volume claim (PVC) with the storageClass specifying the kms key :

Reviewer 4: ```suggestion
. Create a persistent volume claim (PVC) with the storageClass specifying the KMS key :
```

Author: Addressed
